File: components/api/api-modules-javascript/src/main/resources/META-INF/dirigible/modules/src/core/context.ts
/*
 * Copyright (c) 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 * SPDX-License-Identifier: EPL-2.0
 */
/**
 * API Context
 */

const ContextFacade = Java.type("org.eclipse.dirigible.components.api.core.ContextFacade");

export class Context{

	public static get(name: string): object {
		return ContextFacade.get(name);
	};

	public static set(name: string, value: object): void {
		ContextFacade.set(name, value);
	};
}

File: components/api/api-modules-python/src/main/resources/META-INF/dirigible/python-modules/dirigible/core/context.py
# Copyright (c) 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
# All rights reserved. This program and the accompanying materials
# are made available under the terms of the Eclipse Public License v2.0
# which accompanies this distribution, and is available at
# http://www.eclipse.org/legal/epl-v20.html
# SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
# SPDX-License-Identifier: EPL-2.0


import java
ContextFacade = java.type('org.eclipse.dirigible.components.api.core')

class Context:
    @staticmethod
    def get(name):
        return ContextFacade.get(name)

    @staticmethod
    def set(name, value):
        ContextFacade.set(name, value)
